src/plugins/RendererPlugin/index.tsx

```diff
@@ -2,10 +2,14 @@ import { IPlugin } from '../../interfaces/IPlugin';
 import { PluginStore } from '../../PluginStore';
 import { Renderer } from './components/Renderer';
 import ComponentUpdatedEvent from './events/ComponentUpdatedEvent';
+import uuid from './uuid';
 
 export class RendererPlugin implements IPlugin {
   public pluginStore: PluginStore = new PluginStore();
-  private componentMap = new Map<string, Array<React.Component>>();
+  private componentMap = new Map<
+    string,
+    Array<{ component: React.Component; key?: string }>
```

Comment:
**Suggestion (non-blocking)**: Let's declare the type in types folder of `RendererPlugin` and use it from there.